lib/clearcra.config.js

```diff
@@ -0,0 +1,26 @@
+const fs = require("fs");
+const config = require("./clearcra.defaultConfig");
+const { resolveRootPath } = require("./utils");
+
+/**
+ *
+ * @description User could write `clearcra.config.json` to override our config.
+ * @example
+ * clearcra.config.json
+ * {
+ *    removeFiles: ['src/App.css', 'public/robots.txt'],
+ *    addFolders: ['src/hooks']
+ * }
+ */
+function mergeConfigFromUser() {
+  try {
+    let userConfig = fs.readFileSync(resolveRootPath("clearcra.config.json"));
+    userConfig = JSON.parse(userConfig);
+    return userConfig;
+  } catch {}
+}
+
+module.exports = {
+  ...config,
+  ...mergeConfigFromUser(),
+};
```

Comment:
I think your code is very clever and I like it a lot, but I'm having a hard time liking the purpose of this config file.

Imagine a user wanting to remove and add files/folders. The steps they would have to take (in regards to the config files functionality) would actually be more cumbersome than just deleting/adding the files/folders manually:

- Create `clearcra.config.json` -file.
- Add the JSON-object inside (with `removeFiles` and `addFolders`).
- Add all paths in the arrays respectively.
- Making sure the paths are correct.

I believe this process is slower than manually deleting/adding files and folders (if the config file needs to be created manually everytime a user creates a new project)...

Is it maybe intended for more than I can think of?
Is it possible for a user to add this config file globally so they can just run `npx clear-cra` and the config is already set up somewhere? (I don't know. This is why I ask.)...

Reply:
Can't agree more, sorry for I didn't think it straight, I'll delete that logic :)

Reply:
I think global config have 2 ways to achieve:

1. Command Line flags e.g. 
   - `npx create-cra --del-files src/app.js`
   - `npx create-cra --add-folders src/components`

2. Command Line interactive
  
   When User first time use `npx create-cra`, ask him whether to save it as preset?
   - if ture: next time use `npx create-cra` will use preset as default, but can reset preset by select another selection of list - `Reset Preset`.
   - if false: ask him everytime. (Refer to @vue/cli preset prompt).

Another way of thinking:

  How about making this library simpler. Only let it clear the unnecessary files,  not add folders or anything like that, or make another library which is specialized used to add preferable files&folders like `Component` `Context` `CSS Module`.

Reply:
Oh! By the way, I have developed a cli tool also helping CRA stuffs - [Create Cra Template](https://www.npmjs.com/package/create-cra-template).
> It generate app template from your current react app which created by CRA.

Reply:
That is a really cool package you've made there... So when someone cleans up their React app and maybe adds some files/folders as they wish, then run `npx create-cra-template` they can just use that template again and again whenever they create a new project. That is awesome...

Haha... I think it's a shame that we both spent so much time coding, because I realize what you said at the end (making the library simpler) is actually the best way to go, and then separating the other logic to a new package (separation of concerns), oh well, one gains experience, and that in itself is valuable 😄.

It would be really cool to collaborate on a project, but just a heads up, I am really not experienced in this opensource-universe. This is actually my first time doing opensource-stuff, so you would have to guide me along the way. I noticed in your pull-requests, you used terms like "chore:", "feat:", "fix:", and so on... and I'm really not familiar with all those terms. But this would definitely be a fun learning experience for me, so if you are still up for it, I'm very openminded and ready...

Reply:
Haha! Glad you say that. Don't worry about yout, I think we can complement each other, because actually I've just been participate in Open-Source project not long. Your extended our idea which is great, so the steps can be like these: 
1. create a react app normally by create-react-app
2. use `clear-cra` to clear unnecessary files
3. use 'gen-cra' to gen common useful files.
4. use 'create-cra-template' to gen a template and publish for reuse in the future.
How good is it!
I can share many basic experience of open source projects to you. Like 'monorepo' 'testing' 'TDD' 'CICD' 'babel&rollup' 'conventional commits' ...
It's very easy, don't worry about that. We can have a remote meeting if you like. You can open a Zoom room and share me the link and tell me when we meet(https://zoom.us/).